numpy/f2py/crackfortran.py

```diff
@@ -2156,6 +2156,15 @@ def getlincoef(e, xset):  # e = a*x+b ; x in xset
                     m1 = re_1.match(ee)
                 c2 = myeval(ee, {}, {})
                 if (a * 0.5 + b == c and a * 1.5 + b == c2):
+                    # gh-8062: return integers instead of floats if possible.
+                    try:
+                        a = int(a)
+                    except:
+                        pass
+                    try:
+                        b = int(b)
+                    except:
+                        pass
```

Comment:
Bare `except` is an antipattern, as this means that f2py now not only swallows `Ctrl+C` but give different results. This should be `except ValueError` or at least `except Exception`?

Would you mind making a follow-up PR?

Reply:
Sorry, that was my oversight. I can do a follow up if @ianthomas23 prefers.

Reply:
For reasons I don't understand, LGTM caught this, but decided to give a CI tick anyway.

Reply:
@jhelie, I think we might have discussed this before - have we forgotten a config option, or is this working as intended?

Reply:
@eric-wieser, @melissawm: My apologies, I will submit a follow-up PR to change `except` to `except ValueError`. I'll wait until gh-17662 is accepted to avoid having to rebase.

Reply:
You shouldn't need to rebase on that PR, since the lines aren't next to each other - github will sort out the merge automatically I'd expect

Reply:
Say, aren't both `try` / `except` clauses redundant here?
The original values of `a` and `b` are created by `myeval()`, which raises an exception if the return type is not a `float` or `int`:

https://github.com/numpy/numpy/blob/ab22e0076608f4825bf9a3ca1b6a0a6a7a670d14/numpy/f2py/crackfortran.py#L2105-L2109

Reply:
Nice catch